File: gameFramework/src/main/java/at/grueneis/game/framework/code/AndroidInput.java
package at.grueneis.game.framework.code;

import java.util.List;

import android.content.Context;
import android.view.View;
import at.grueneis.game.framework.Input;

public class AndroidInput implements Input
{
	SensorHandler sensorHandler;
	TouchHandler touchHandler;
	
	public AndroidInput(Context context, View view, float scaleX, float scaleY)
	{
		sensorHandler = new SensorHandler(context);
		//keyHandler = new KeyboardHandler(view);
		//touchHandler = new SingleTouchHandler(view, scaleX, scaleY);
		touchHandler = new MultiTouchHandler(view, scaleX, scaleY);
	}
	
	public boolean isTouchDown(int pointer)
	{
		return touchHandler.isTouchDown(pointer);
	}
	
	public int getTouchX(int pointer)
	{
		return touchHandler.getTouchX(pointer);
	}
	
	public int getTouchY(int pointer)
	{
		return touchHandler.getTouchY(pointer);
	}
	
	public float getAccelX()
	{
		return sensorHandler.getGravityX();
	}
	
	public float getAccelY()
	{
		return sensorHandler.getGravityY();
	}
	
	public float getAccelZ()
	{
		return sensorHandler.getGravityZ();
	}
	
	public List<TouchEvent> getTouchEvents()
	{
		return touchHandler.getTouchEvents();
	}

	public int getPointerCount() {
		return touchHandler.getPointerCount();
	}
	
}

File: gameFramework/src/main/java/at/grueneis/game/framework/impl/AndroidInput.kt
package at.grueneis.game.framework.impl

import android.content.Context
import android.view.View
import at.grueneis.game.framework.Input
import at.grueneis.game.framework.Input.TouchEvent

class AndroidInput(context: Context, view: View, scaleX: Float, scaleY: Float) : Input {
    var sensorHandler: SensorHandler = SensorHandler(context)
    var touchHandler: TouchHandler = MultiTouchHandler(view, scaleX, scaleY)
    override fun isTouchDown(pointer: Int): Boolean {
        return touchHandler.isTouchDown(pointer)
    }

    override fun getTouchX(pointer: Int): Int {
        return touchHandler.getTouchX(pointer)
    }

    override fun getTouchY(pointer: Int): Int {
        return touchHandler.getTouchY(pointer)
    }

    override val accelX: Float
        get() = sensorHandler.gravityX
    override val accelY: Float
        get() = sensorHandler.gravityY
    override val accelZ: Float
        get() = sensorHandler.gravityZ
    override val touchEvents: List<TouchEvent>?
        get() = touchHandler.touchEvents
    override val pointerCount: Int
        get() = touchHandler.pointerCount

    init {
        //keyHandler = new KeyboardHandler(view);
        //touchHandler = new SingleTouchHandler(view, scaleX, scaleY);
    }
}